elasticdl/python/common/k8s_job_monitor.py

```diff
@@ -0,0 +1,199 @@
+            ps_num: Integer, parameter server number of the job.
+        """
+        self.worker_num = worker_num
+        self.ps_num = ps_num
+        self.job_name = job_name
+        self.client = Client(
+            image_name=None,
+            namespace=namespace,
+            job_name=job_name,
+            force_use_kube_config_file=True,
+        )
+
+    def check_worker_status(self):
+        for i in range(self.worker_num):
+            worker_pod = self.client.get_worker_pod(i)
+            worker_pod_name = self.client.get_worker_pod_name(i)
+            if worker_pod is None:
+                print("Worker {} Not Found".format(worker_pod_name))
+            elif worker_pod.status.phase == PodStatus.FAILED:
+                logger.error(
+                    "Worker {} {}".format(
+                        worker_pod_name, worker_pod.status.phase
+                    )
+                )
+                print("{} log :".format(worker_pod_name))
+                log = self.client.get_worker_log(i)
+                print_tail_log(log, tail_num=100)
+
+    def check_ps_status(self):
+        for i in range(self.ps_num):
+            ps_pod = self.client.get_ps_pod(i)
+            ps_pod_name = self.client.get_ps_pod_name(i)
+            if ps_pod is None:
+                print("PS {} Not Found".format(ps_pod_name))
+            elif ps_pod.status.phase == PodStatus.FAILED:
+                print("PS {} {}".format(ps_pod_name, ps_pod.status.phase))
+
+    def show_master_increment_log(self, old_log):
+        new_log = self.client.get_master_log()
+        if new_log is None:
+            return
+        increment_log = new_log.replace(old_log, "")
+        task_log = ""
+        for log_line in increment_log.split("\n"):
+            if "Evaluation" in log_line:
+                print(log_line)
+            if "Task" in log_line:
+                task_log = log_line
+        print(task_log)
+        return new_log
+
+    def monitor_status(self):
+        retry_num = 0
+        job_success = False
+        master_log = ""
+        while True:
+            master_pod = self.client.get_master_pod()
+            if master_pod is None:
+                retry_num += 1
+                if retry_num > MAX_READ_POD_RETRIES:
+                    raise ValueError(
+                        "{} Not Found".format(
+                            self.client.get_master_pod_name()
+                        )
+                    )
+                print("{} Not Found".format(self.client.get_master_pod_name()))
+                time.sleep(10)
+                continue
+
+            print("Master status: {}".format(master_pod.status.phase))
+            if master_pod.status.phase == PodStatus.SUCCEEDED:
+                job_success = True
+                break
+            elif master_pod.status.phase in [
+                PodStatus.PENDING,
+                PodStatus.CREATING,
+            ]:
+                time.sleep(120)
```

Comment:
Are you sure? This is very long time

Reply:
The job execution job varies from minutes to hours. And I have reduced it to 60s. It is difficult to set the interval.

Reply:
Sure but this is only the pod pending/creating time instead of the job execution time

Reply:
Oh, I have mixed the two intervals. May 10s be enough?

Reply:
This depends on the cluster.

Reply:
Set a default value and the user can set it in arguments?